### ISSUE TITLE
Android: Intl.NumberFormat style percent not working properly

### ISSUE THREAD

**Issue by Bekaxp** [facebook/hermes#1138]
## Bug Description
When trying to format a number as a percent with the following code:

```
Intl.NumberFormat('en', {style: 'percent'}).format(0.1091);
```

the output on Android is incorrect. Instead of `%` sign a `+` sign is displayed. This works correctly for the web and iOS. Please have a look at the repro bellow.

React Native version (if any): 0.72.4 (Expo SDK 49)
OS version (if any): Android API level 34

## Steps To Reproduce
The following code:

```
import { Text, SafeAreaView } from 'react-native';

export default function App() {
  return (
    <SafeAreaView style={{flex: 1, justifyContent: 'center', alignContent: 'center'}}>
      <Text style={{textAlign: 'center', fontSize: 24}}>
        {Intl.NumberFormat('en', {
          style: 'percent',
          minimumFractionDigits: 2,
          signDisplay: 'exceptZero'
        }).format(0.1091)}
      </Text>
    </SafeAreaView>
  );
}
```

will produce these results:

**Android**
<img width="322" alt="Screenshot 2023-10-01 at 11 11 22" src="https://github.com/facebook/hermes/assets/13329079/168c6c42-e332-44cc-bf16-31f2d022c6ee">

**iOS**
<img width="325" alt="Screenshot 2023-10-01 at 11 11 45" src="https://github.com/facebook/hermes/assets/13329079/e24ed91b-9098-4478-8317-b65ec9a0adb4">

As you can see, on Android the`%` sign is replaced by `+`, which is incorrect.

### Snack example
[SNACK LINK](https://snack.expo.dev/PkoMc6ByS)

## The Expected Behavior
Android should display the `%` sign correctly when formatting a number with `Intl.NumberFormat` and `style: 'percent'`.

**Comment by tamlyn** [comment]
Related to – or possibly duplicate of – https://github.com/facebook/hermes/issues/789

Pre-Hermes we were using the [Intl polyfills](https://formatjs.io/docs/polyfills/) which worked nicely. But now it seems the polyfills are deferring to the native broken implementation. Any way to force the polyfill to be used?

In the meantime it's best not to use `signDisplay`. Luckily it's quite easy to implement the behaviour yourself.

**Comment by shubhamguptadream11** [comment]
I had fixed this issue here: https://github.com/facebook/hermes/pull/1483. Please try it once

**Comment by Bekaxp** [comment]
> I had fixed this issue here: #1483. Please try it once

Awesome! Thanks! I can apply a patch with your fix and try it out but hopefully it gets merged soon, then we have it 👍 .

**Comment by shubhamguptadream11** [comment]
Fixed PR has now been merged. We can close this issue now as resolved.

**Comment by Bekaxp** [comment]
Nice, thanks, so it should be already available with RN 0.76 release which will happen soon?
